src/ratelimit/RateLimitController.ts

```diff
@@ -169,13 +169,15 @@ export class RateLimitController<
    */
   private recordRequest(api: keyof RateLimitedApis, origin: string) {
     this.update((state) => {
-      (state as any).requests[api][origin] =
-        ((state as any).requests[api][origin] ?? 0) + 1;
-
-      setTimeout(
-        () => this.resetRequestCount(api, origin),
-        this.rateLimitTimeout,
-      );
+      const previous = (state as any).requests[api][origin] ?? 0;
+      (state as any).requests[api][origin] = previous + 1;
+
+      if (previous === 0) {
+        setTimeout(
+          () => this.resetRequestCount(api, origin),
+          this.rateLimitTimeout,
+        );
+      }
```

Comment:
If the intended behavior is that the first request initiates a timeout, during which the requesting origin can make no more than `rateLimitCount` number of requests, then my late-night brain assesses that this is correct.

Previously, every request would initiate a new timeout, regardless of whether the previous one had expired, and that's definitely not correct.

Reply:
That's exactly the conclusion that I came to as well, your described behaviour was the intention, but it may been altered throughout the implementation process without a good test to remind me. That should be corrected with this PR. Feel free to wait for tomorrow to do an actual review 😄